test_automation/conftest.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+import subprocess
+import time
+import os
+import sys
+import argparse
+import random
+import json
+import logging
+
+import config.config as pconfig
+import utility.logger as plogger
+from service_client.generic import GenericServiceClient
+import crypto.crypto as crypto
+import utility.signature as signature
+import worker.worker_details as worker
+from shared_kv.shared_kv_interface import KvStorage
+import utility.utility as enclave_helper
+import utility.file_utils as futils
+from process_request import process_request
+
+TCFHOME = os.environ.get("TCF_HOME", "../../")
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_config(args=None):
+    """ Fixture to setup initial config for pytest session. """
+
+    # parse out the configuration file first
+    conffiles = [ "tcs_config.toml" ]
+    confpaths = [ ".", TCFHOME + "/config", "../../etc"]
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", help="configuration file", nargs = "+")
+    parser.add_argument("--config-dir", help="configuration folder", nargs = "+")
+    (options, remainder) = parser.parse_known_args(args)
+
+    if options.config :
+        conffiles = options.config
+
+    if options.config_dir :
+        confpaths = options.config_dir
+
+    try :
+        config = pconfig.parse_configuration_files(conffiles, confpaths)
+        config_json_str = json.dumps(config, indent=4)
+    except pconfig.ConfigurationException as e :
+        logger.error(str(e))
+        sys.exit(-1)
+
+    plogger.setup_loggers(config.get("Logging", {}))
+    sys.stdout = plogger.stream_to_logger(logging.getLogger("STDOUT"), logging.DEBUG)
+    sys.stderr = plogger.stream_to_logger(logging.getLogger("STDERR"), logging.WARN)
+
+    sig_obj, worker_obj, private_key, server_uri = initialize_objects(config, remainder)
+    worker_obj, uri_client, err_cd = worker_lookup_retrieve(config, worker_obj, server_uri)
+
+    return worker_obj, sig_obj, uri_client, private_key, err_cd
+
+def initialize_objects(config, args):
+    """ Function to initialize common objects for tests. """
+
+    logger.info('***************** INTEL TRUSTED COMPUTE FRAMEWORK (TCF)*****************')
+
+    # static server_uri and private_key of client
+    server_uri = r'http://localhost:1947'
```

Comment:
A Python `r''` string is only needed if you have backslashes  (`\`).  Remove the `r`.

Reply:
Removed the unnecessary r'' and added the value to default for command line option, to provide functionality to test scripts to use different uri that default localhost